model/PDOFactory.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace EmilieSchott\BlogPHP\Model;
+
+class PDOFactory {
+    private static $db;
+
+    public static function getDbConnection() {
+        if (self::$db instanceof \PDO) {
+            return self::$db;
+        } else {
+            self::$db = new \PDO('mysql:host=localhost;dbname=p5_blog_php;port=3308;charset=utf8', 'root', '');
+            return self::$db;
+        }
+    }
```

Comment:
No need to refactor this yet, but just so you know...
If you want to optimize a bit, drop the "else" part. It will work just the same (since we return the result if we enter the "if") and prevent PHP from calculating an unnecessary pathway.
```suggestion
    public static function getDbConnection() {
        if (self::$db instanceof \PDO) {
            return self::$db;
        }
        self::$db = new \PDO('mysql:host=localhost;dbname=p5_blog_php;port=3308;charset=utf8', 'root', '');
        return self::$db;
    }
```

Reply:
Ok, interresting. Thank you to let met know. I'm going to make the change and tell you when it's done.